samples/Jzon.cpp

```diff
@@ -373,76 +374,70 @@ namespace Jzon
 		return new Value(*this);
 	}
 
-	// This is not the most beautiful place for these, but it'll do
-	static const char charsUnescaped[] = { '\\'  , '/'  , '\"'  , '\n' , '\t' , '\b' , '\f' , '\r' };
-	static const char *charsEscaped[]  = { "\\\\", "\\/", "\\\"", "\\n", "\\t", "\\b", "\\f", "\\r" };
-	static const unsigned int numEscapeChars = 8;
-	static const char nullUnescaped = '\0';
-	static const char *nullEscaped  = "\0\0";
-	const char *&getEscaped(const char &c)
-	{
-		for (unsigned int i = 0; i < numEscapeChars; ++i)
-		{
-			const char &ue = charsUnescaped[i];
+    // This is not the most beautiful place for these, but it'll do
+    constexpr std::array<char, 8> charsUnescaped = {'\\', '/', '\"', '\n', '\t', '\b', '\f', '\r'};
+    constexpr std::array<const char *, 8> charsEscaped = {"\\\\", "\\/", "\\\"", "\\n", "\\t", "\\b", "\\f", "\\r"};
+    constexpr char nullUnescaped = '\0';
+    constexpr const char *nullEscaped = "\0\0";
 
-			if (c == ue)
-			{
-				const char *&e = charsEscaped[i];
-				return e;
-			}
-		}
-		return nullEscaped;
-	}
-	const char &getUnescaped(const char &c1, const char &c2)
-	{
-		for (unsigned int i = 0; i < numEscapeChars; ++i)
-		{
-			const char *&e = charsEscaped[i];
+    const char *const &getEscaped(const char &c)
+    {
+        for (unsigned int i = 0; i < charsEscaped.size(); ++i) {
+            const char &ue = charsUnescaped[i];
 
-			if (c1 == e[0] && c2 == e[1])
-			{
-				const char &ue = charsUnescaped[i];
-				return ue;
-			}
-		}
-		return nullUnescaped;
-	}
+            if (c == ue) {
+                const auto *const &e = charsEscaped[i];
+                return e;
+            }
+        }
+        return nullEscaped;
+    }
 
-	std::string Value::EscapeString(const std::string &value)
-	{
+    const char &getUnescaped(const char &c1, const char &c2)
+    {
+        for (unsigned int i = 0; i < charsEscaped.size(); ++i) {
+            const auto *const &e = charsEscaped[i];
+
+            if (c1 == e[0] && c2 == e[1]) {
+                const auto &ue = charsUnescaped[i];
+                return ue;
+            }
+        }
+        return nullUnescaped;
+    }
+
+    std::string Value::EscapeString(const std::string &value)
+    {
         return std::accumulate(value.begin(), value.end(), std::string(""), [](std::string r, char c) {
-            const char *&a = getEscaped(c);
+            auto *const &a = getEscaped(c);
```

Comment:
Honestly, here I prefer either to have the original code or simply use `auto`.  `auto *const &` looks for me less clear than the original code 👀

Reply:
Looking at this closely,

```
const auto *&a
```

evaluates to 

```
const char *&a
```

which is incorrect.

I think

```
const auto &a
```

will evaluate to the correct variable type.

```
auto &a
```

doesn't sound like it will.

Reply:
Decided to just get rid of some of these.

Reply:
Good, I did some tests on my side and effectively `const auto &a ` is the better option 👍